### ISSUE TITLE
compileDebugJavaWithJavac fails with NoAnnotatedMethodsException when using Kotlin

### ISSUE THREAD

**Issue by fdorssers** [hotchemi/PermissionsDispatcher#112]
It seems that the fix to issue #109 either introduced a new issue, or has only just made it possible to actually reach this problem.

```
:app:compileDebugKotlin UP-TO-DATE
:app:compileDebugJavaWithJavac FAILED
Error:Execution failed for task ':app:compileDebugJavaWithJavac'.
> permissions.dispatcher.processor.exception.NoAnnotatedMethodsException: Annotated class 'MainActivity' doesn't have any method annotated with '@NeedsPermission'
```

This happens with Kotlin version 1.0.0, Android Studio 2.0 beta 6, Gradle version 2.0.0 beta 6 and JDK 1.8.0 update 65. No idea if it's the same with other versions.

These are the relevant lines from the gradle file (updated to 2.0.8):
```gradle
compile 'com.github.hotchemi:permissionsdispatcher:2.0.8'
kapt 'com.github.hotchemi:permissionsdispatcher-processor:2.0.8'
```

And this is the activity:
```kotlin
@RuntimePermissions
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById(R.id.btn_camera).setOnClickListener {
            MainActivityPermissionsDispatcher.showCameraWithCheck(this)
        }
    }

    @NeedsPermission(Manifest.permission.CAMERA)
    fun showCamera() {
        Toast.makeText(this, "Showing camera", Toast.LENGTH_SHORT).show()
    }

    @OnShowRationale(Manifest.permission.CAMERA)
    fun showRationaleForCamera(request: PermissionRequest) {
        AlertDialog.Builder(this)
                .setMessage("Requesting camera")
                .setPositiveButton("Allow", { dialog, button -> request.proceed() })
                .setNegativeButton("Deny", { dialog, button -> request.cancel() })
                .show()
    }

    @OnPermissionDenied(Manifest.permission.CAMERA)
    fun showDeniedForCamera() {
        Toast.makeText(this, "Camera denied", Toast.LENGTH_SHORT).show()
    }

    @OnNeverAskAgain(Manifest.permission.CAMERA)
    fun showNeverAskForCamera() {
        Toast.makeText(this, "Camera always denied", Toast.LENGTH_SHORT).show()
    }
}
```

Steps to reproduce:
- Clone [this](https://github.com/fdorssers/kotlin-permissions) repository
- Open with Android Studio
- Rebuild
- Run

Interestingly enough the previous issue gave the error during rebuilding, while this one only throws it once you actually try to run it.

**Comment by hotchemi** [comment]
Umm wired...thx for your report.

**Comment by hotchemi** [comment]
refs. https://github.com/hotchemi/PermissionsDispatcher/blob/e0dde623c415b666f3de6ee0b56daed52d2d4943/processor/src/main/kotlin/permissions/dispatcher/processor/util/Validators.kt#L56

**Comment by shiraji** [comment]
I think the problem is kapt. Realm has the same issue with this. Check their issue. https://github.com/realm/realm-java/issues/509

So, for now, according to them, we need to change annotation's retention `@Retention(RetentionPolicy.CLASS)`.

I could not find good documentation for this issue except in Japanese blog post http://sys1yagi.hatenablog.com/entry/2016/02/02/130402 (Thanks @sys1yagi)

**Comment by hotchemi** [comment]
@shiraji Thx!

**Comment by hotchemi** [comment]
This issue might be fixed in master branch so I'll test it tonight and release new version.

**Comment by fdorssers** [comment]
Did you already happen to get around to testing the possible fix?

**Comment by hotchemi** [comment]
Sorry I forgot to do that, do it soon.

**Comment by hotchemi** [comment]
@fdorssers Just released 2.0.9. Sorry for the late.
